src/main/java/com/thealgorithms/datastructures/heaps/LeonardoHeap.java

```diff
@@ -0,0 +1,225 @@
+package com.thealgorithms.datastructures.heaps;
+
+import com.thealgorithms.bitmanipulation.SingleBitOperations;
+import com.thealgorithms.maths.LeonardoNumber;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Wikipedia: https://en.wikipedia.org/wiki/Smoothsort
+ */
+public class LeonardoHeap<T extends Comparable<T>> {
+
+    private int leonardoLevelTracker;
+    private int leonardoHeapSize;
+    private final List<T> leonardoHeap;
```

Comment:
```suggestion
    private int leonardoLevelTracker = 0;
    private int leonardoHeapSize = 0;
    private final List<T> leonardoHeap = new ArrayList<T>();
```

Furthermore:
- is that true that `leonardoHeapSize == leonardoHeap.size()`? If yes, I would remove `leonardoHeapSize`,
- please consider renaming these members - the `leonardo` prefix is redundant.